### ISSUE TITLE
[REVIEW]: Sapsan: Framework for Supernovae Turbulence Modeling with Machine Learning

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3199]
**Submitting author:** @pikarpov-lanl (<a href="http://orcid.org/0000-0003-4311-8490">Platon I. Karpov</a>)
**Repository:** <a href="https://github.com/pikarpov-LANL/Sapsan" target ="_blank">https://github.com/pikarpov-LANL/Sapsan</a>
**Version:** v0.4.6
**Editor:** @dfm
**Reviewer:** @kburns, @milescranmer
**Archive:** <a href="https://doi.org/10.5281/zenodo.5720254" target="_blank">10.5281/zenodo.5720254</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/4967669457d68b7fd38bc4e4c8fecff7/status.svg)](https://joss.theoj.org/papers/4967669457d68b7fd38bc4e4c8fecff7)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/4967669457d68b7fd38bc4e4c8fecff7"><img src="https://joss.theoj.org/papers/4967669457d68b7fd38bc4e4c8fecff7/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/4967669457d68b7fd38bc4e4c8fecff7/status.svg)](https://joss.theoj.org/papers/4967669457d68b7fd38bc4e4c8fecff7)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/pikarpov-LANL/Sapsan" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@kburns & @milescranmer, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @dfm know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pikarpov-LANL/Sapsan">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@pikarpov-lanl) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pikarpov-LANL/Sapsan">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@pikarpov-lanl) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @kburns, @milescranmer it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.20 s (349.6 files/s, 38864.2 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          57           1268           1307           4174
Markdown                         4             77              0            203
Jupyter Notebook                 3              0            492            166
YAML                             5             12              3            161
Dockerfile                       1              5              3             11
make                             1              2              3              5
-------------------------------------------------------------------------------
SUM:                            71           1364           1808           4720
-------------------------------------------------------------------------------


Statistical information for the repository '2b39743c09c044cc12370bcb' was
gathered on 2021/04/21.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
IceKhan13                        1            43              6            0.21
Iskandar Sitdikov               22          3330            396           16.31
Platon I Karpov                 16          3825           2270           26.68
Platon Karpov                   16          1544           1217           12.09
pikarpov                        57          6688           3522           44.70

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
IceKhan13                     6           14.0         16.5                0.00
Iskandar Sitdikov          1493           44.8         12.3                7.97
Platon I Karpov            2814           73.6          4.7               12.08
Platon Karpov              2436          157.8          4.7                6.65
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3199 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by dfm** [comment]
@kburns, @MilesCranmer, @pikarpov – This is the review thread for the paper. All of our communications will happen here from now on. Thanks again for agreeing to participate!

Please read the "Reviewer instructions & questions" in the first comment above.

Both reviewers have checklists at the top of this thread (in that first comment) with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention openjournals/joss-reviews#3199 so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for the review process to be completed within about 4-6 weeks but please make a start well ahead of this as JOSS reviews are by their nature iterative and any early feedback you may be able to provide to the author will be very helpful in meeting this schedule.

**Comment by dfm** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:wave: @kburns, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @milescranmer, please update us on how your review is going (this is an automated reminder).

**Comment by MilesCranmer** [comment]
Planning on sharing my initial comments by this weekend!

Cheers,
Miles

**Comment by dfm** [comment]
:wave: @kburns, @MilesCranmer - Any updates on this? Let me know if you have any questions or if you run into any issues. Thanks!

**Comment by kburns** [comment]
@dfm Thanks for the reminder and sorry for the delay. I've posted my comments and questions in https://github.com/pikarpov-LANL/Sapsan/issues/99.

**Comment by MilesCranmer** [comment]
Hi all, sorry for the long delay - this fell off my radar during the annual NeurIPS submission chaos. Here is my initial round of comments: https://github.com/pikarpov-LANL/Sapsan/issues/100.

Cheers,
Miles

**Comment by pikarpov-LANL** [comment]
Hi everyone! First and foremost, thank you for your comments. I have made appropriate edits and posted my responses right in the Issues thread. 

@MilesCranmer you can check them out [pikarpov-LANL/Sapsan#100](https://github.com/pikarpov-LANL/Sapsan/issues/100)

@kburns I am finishing up the response and will post it here tonight [pikarpov-LANL/Sapsan#99](https://github.com/pikarpov-LANL/Sapsan/issues/99)

**Comment by pikarpov-LANL** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: :point_left:

**Comment by dfm** [comment]
@MilesCranmer, @kburns, @pikarpov-LANL: I wanted to check in here to see where we stand. It looks to me like we're currently waiting on @MilesCranmer and @kburns to take a look at @pikarpov-LANL's responses on the review issues (https://github.com/pikarpov-LANL/Sapsan/issues/99, https://github.com/pikarpov-LANL/Sapsan/issues/100). Is that correct and, if so, @MilesCranmer, @kburns can you make sure that you keep this on your radar and let me know if you have issues or questions? Thanks all!!

**Comment by pikarpov-LANL** [comment]
@dfm indeed! I have responded to the comments and have fixed up all the noted issues in the code.

**Comment by dfm** [comment]
@MilesCranmer, @kburns: Do you have any updates on how your reviews are going? Thanks!

**Comment by kburns** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: :point_left:

**Comment by kburns** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: :point_left:

**Comment by kburns** [comment]
@dfm my checklist is complete. Thanks to the authors for their patience! I think everything looks great except some superscripts don't seem to be rendering properly in the PDF, (e.g. line 131 in the latest proof).

**Comment by MilesCranmer** [comment]
Hi @dfm @pikarpov-LANL and co-authors,

I sincerely apologize for the late followup. I have been on an internship where all the code writing is internal, and have been bad at going through my GitHub notifications, and this dropped off my radar. I am very sorry about that, and thanks for your patience! (and thanks @dfm for the helpful ping :) )

All of my comments have been addressed - see also my response in the issue - and all checkpoints above are hit. I am very happy with the Sapsan project! It's quite a nice framework, and the plethora of documentation available is fantastic - perfect for the intended audience(s).

Cheers,
Miles

**Comment by dfm** [comment]
@kburns, @MilesCranmer — Thanks for your constructive reviews! I really appreciate the time that you took to contribute.

@pikarpov-LANL — Thanks for all your work responding to these reviews and for your submission! Give me a few days to do some final edits and then I'll have a couple of last tasks for you before publication.

**Comment by dfm** [comment]
@pikarpov-LANL — I've opened a pull request with some edits to the paper. Can you take a look at that and once you've merged and fixed the remaining issues, please take the following steps:

1. Comment `@whedon generate pdf from branch joss` on this thread and read through the manuscript to make sure that you're happy with it (it's hard to make changes later!), especially your name and affiliation.
2. Increment the version number of the software and report that version number back here.
3. Create an archived release of that version of the software (using Zenodo or something similar). Please make sure that the metadata (title and author list) exactly match the paper. Then report the DOI of the release back to this thread.

Let me know if you have questions or run into any issues!

**Comment by pikarpov-LANL** [comment]
@dfm thank you for the edits! I reviewed and accepted the pull request. I would like to update the the figures a little bit to be closer to the plots from my Astrophysical Journal paper; text will remain the same. Let me just run the final version by my collaborators, and then I will go through the last steps you outlined.

**Comment by dfm** [comment]
@pikarpov-LANL: great! So, is there a parallel ApJ paper to this one? What is the current status of that review?

**Comment by pikarpov-LANL** [comment]
@dfm Yes, there is. We are about to submit it for review; I will include the appropriate reference. It barely includes any details on `Sapsan` itself, besides mentioning it. Results/plot wise the paper is much more extensive; JOSS paper just includes a few examples.

**Comment by pikarpov-LANL** [comment]
@dfm I just pushed the latest version of Sapsan

1. I need to update the plots to reflect the latest changes in analytical subgrid models included with Sapsan + latest formatting. I will get back to you next week, as I am currently on travel.
2. Sapsan v0.4.5
3. DOI: 10.5281/zenodo.5587205

**Comment by dfm** [comment]
@pikarpov-LANL: You said

> I will get back to you next week, as I am currently on travel.

a few weeks ago - any updates on this? I know how the weeks seem to go these days, but I wanted to check in :D

**Comment by pikarpov-LANL** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03199/joss.03199/10.21105.joss.03199.pdf) :page_facing_up: :point_left:

**Comment by pikarpov-LANL** [comment]
@dfm , thank you for checking in! Unfortunately, due to a family emergency on the other side of the globe, I had to spend a few weeks with my family. This caused a number of delays on all work fronts, but I just got back and trying to catch up.

That said, I think that I am happy with the paper at the point. All of the figures have been updated to reflect the latest state of Sapsan. In addition, the examples in the paper better showcase various capabilities of dealing with 2D and 3D data, numerous analytical tools, and compatibility with different ML libraries. Lastly, I made a few minor tweaks and have pushed the newest version of Sapsan. Here is the updated info:

Latest version: Sapsan v0.4.6
DOI: 10.5281/zenodo.5720254

Please let me know if there is anything else that I can do.

**Comment by dfm** [comment]
@whedon check references from branch joss

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch joss
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1103/PhysRevE.93.031301 is OK
- 10.1103/PhysRevFluids.3.074602 is OK
- 10.1103/RevModPhys.91.045002 is OK
- 10.1080/14685240802376389 is OK
- 10.1109/tkde.2017.2720168 is OK
- 10.1038/nature12128 is OK
- 10.1126/science.359.6377.725 is OK
- 10.1017/S0022112094002296 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by dfm** [comment]
@whedon set v0.4.6 as version

**Comment by whedon** [comment]
OK. v0.4.6 is the version.

**Comment by dfm** [comment]
@whedon set 10.5281/zenodo.5720254 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5720254" target="_blank">10.5281/zenodo.5720254</a> is the archive.

**Comment by dfm** [comment]
@whedon recommend-accept from branch joss

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1103/PhysRevE.93.031301 is OK
- 10.1103/PhysRevFluids.3.074602 is OK
- 10.1103/RevModPhys.91.045002 is OK
- 10.1080/14685240802376389 is OK
- 10.1109/tkde.2017.2720168 is OK
- 10.1038/nature12128 is OK
- 10.1126/science.359.6377.725 is OK
- 10.1017/S0022112094002296 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2763

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2763, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss 
```

**Comment by dfm** [comment]
@pikarpov-LANL: I've passed this off to the EiCs, who will handle the final processing (and may have some last edits). Thanks for your submission and all your hard work through the process!

@kburns, @MilesCranmer: Thanks both for your reviews! I really appreciate the time you volunteered and all the work you put into your reviews! 🎉

**Comment by MilesCranmer** [comment]
Thanks @dfm. Congrats on the JOSS paper @pikarpov-LANL!

**Comment by Kevin-Mattheus-Moerman** [comment]
I proofread the paper and reviewed the ZENODO archive. All seems in order. I will now proceed to accept this work in JOSS.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true from branch joss

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1464232461710286852) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2769
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03199](https://doi.org/10.21105/joss.03199)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by Kevin-Mattheus-Moerman** [comment]
Looks like we are all set. 

@pikarpov-LANL Congratulations on getting this work published in JOSS! 

Thanks a lot @kburns, @MilesCranmer for your stellar review efforts! Thanks @dfm for editing this work!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03199/status.svg)](https://doi.org/10.21105/joss.03199)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03199">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03199/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03199/status.svg
   :target: https://doi.org/10.21105/joss.03199
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03199/status.svg)](https://doi.org/10.21105/joss.03199)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by pikarpov-LANL** [comment]
@dfm , @MilesCranmer , @kburns , @Kevin-Mattheus-Moerman - thank you all for helping improve Sapsan framework and the paper. The review process itself has been immensely helpful for myself and my collaborators. This publication will make it much easier to promote Sapsan in the community. Lastly, it will be proudly highlighted in my upcoming dissertation!
